### PR TITLE
New PutMedia API

### PR DESCRIPTION
**Pull request by at-wat**
https://github.com/seqsense/kinesisvideomanager/issues/307
Switch to Reader/Writer pattern.

### CODE REVIEW

File: examples/putmedia/main.go [hunk 1]
```diff
@@ -94,19 +107,13 @@ func main() {
 			},
 		}
 		// log.Printf("write: %v", data)
-		ch <- data
+		err := w.Write(data)
+		if err != nil {
+			log.Println(err)
+		}
 	})
 	defer as.Close()
```
Comment (kamatama41) [inline examples/putmedia/main.go:115]: `w` (BlockWriter) should be closed as well.

```go
	defer func() {
		as.Close()
		w.Close()
	}()
```

File: provider.go [hunk 2]
```diff
@@ -48,7 +48,10 @@ var (
 	regexAmzCredHeader = regexp.MustCompile(`X-Amz-(Credential|Security-Token|Signature)=[^&]*`)
 )
 
-var ErrInvalidTimecode = errors.New("invalid timecode")
+var (
+	ErrInvalidTimecode = errors.New("invalid timecode")
+	ErrWriteTimeout    = errors.New("write timeout")
```
Comment (kamatama41) [inline provider.go:53]: The error seems not to be currently used. Is this necessary?

File: provider.go [hunk 3]
```diff
@@ -91,13 +94,37 @@ func (c *Client) Provider(streamID StreamID, tracks []TrackEntry) (*Provider, er
 }
 
 type BlockWriter interface {
+	// Write a block to Kinesis Video Stream.
 	Write(*BlockWithBaseTimecode) error
+	// Read a response from Kinesis Video Stream.
```
Comment (kamatama41) [inline provider.go:99]: Follow the godoc style.
```suggestion
	// ReadResponse reads a response from Kinesis Video Stream.
```

File: provider.go (hunk 2 above)
Comment (at-wat) [reply provider.go:53]: https://github.com/seqsense/kinesisvideomanager/blob/9035d1dfb54853f89c65d088a4bfe7137f238923/provider.go#L296-L298
should be `ErrWriteTimeout`

**Comment by at-wat** [comment]
@kamatama41 addressed the comments. PTAL

**Review by kamatama41 — APPROVED** [review]
LGTM. Let's try 🚀

**Pull request merged**